Objects/TI4_Helpers/TI4_TechnologyHelper.ttslua

```diff
@@ -0,0 +1,174 @@
+	["Exotrireme II"] = { type = "unitUpgrade", requirements = "BBY", abbrev = 'Exotrireme II', expansion = "base" },
+	["Valkyrie Particle Weave"] = { type = "Red", requirements = "RR", abbrev = 'Valkyrie PW', expansion = "base" },
+	["Spacial Conduit Cylinder"] = { type = "Blue", requirements = "BB", abbrev = 'Spacial Conduit', expansion = "base" },
+	["E-res Siphons"] = { type = "Yellow", requirements = "YY", abbrev = 'E-Res', expansion = "base" },
+	["Lazax Gate Folding"] = { type = "Blue", requirements = "BB", abbrev = 'Lazax Gate', expansion = "base" },
+	["Hegemonic Trade Policy"] = { type = "Yellow", requirements = "YY", abbrev = 'Hegemonic', expansion = "base" },
+	["Instinct Training"] = { type = "Green", requirements = "G", abbrev = 'Instinct Train', expansion = "base" },
+	["Nullification Field"] = { type = "Yellow", requirements = "YY", abbrev = 'Null. Field', expansion = "base" },
+	["Yin Spinner"] = { type = "Green", requirements = "GG", abbrev = 'Yin Spin', expansion = "base" },
+	["Impulse Core"] = { type = "Yellow", requirements = "YY", abbrev = 'Impulse', expansion = "base" },
+	["Mageon Implants"] = { type = "Green", requirements = "GGG", abbrev = 'Mageon', expansion = "base" },
+	["Transparasteel Plating"] = { type = "Green", requirements = "G", abbrev = 'Transparasteel', expansion = "base" },
+	-- PoK faction tech
+	["Temporal Command Suite"] = { type = "Yellow", requirements = "Y", abbrev = 'Temp Cmd Suite', expansion = "base" },
+	["Memoria II"] = { type = "unitUpgrade", requirements = "GBY", abbrev = 'Memoria II', expansion = "base" },
+	["Vortex"] = { type = "Red", requirements = "R", abbrev = 'Vortex', expansion = "base" },
+	["Dimensional Tear II"] = { type = "unitUpgrade", requirements = "YY", abbrev = 'Dim Tear II', expansion = "base" },
+	["Aerie Hololattice"] = { type = "Yellow", requirements = "Y", abbrev = 'Aerie Holo', expansion = "base" },
+	["Strike Wing Alpha II"] = { type = "unitUpgrade", requirements = "RR", abbrev = 'String Wing II', expansion = "base" },
+	["Saturn Engine II"] = { type = "unitUpgrade", requirements = "GYR", abbrev = 'Sat Eng II', expansion = "base" },
+	["Hel Titan II"] = { type = "unitUpgrade", requirements = "YR", abbrev = 'Hel-Titan II', expansion = "base" },
+	["Aetherstream"] = { type = "Blue", requirements = "BB", abbrev = 'Aetherstream', expansion = "base" },
+	["Voidwatch"] = { type = "Green", requirements = "G", abbrev = 'Voidwatch', expansion = "base" },
+	["Crimson Legionnaire II"] = { type = "unitUpgrade", requirements = "GG", abbrev = 'Crimson Legin II', expansion = "base" },
+	["Genetic Recombination"] = { type = "Green", requirements = "G", abbrev = 'Gene Recomb', expansion = "base" },
+	["Supercharge"] = { type = "Red", requirements = "R", abbrev = 'Supercharge', expansion = "base" },
+	["Pre-Fab Arcologies"] = { type = "Green", requirements = "GGG", abbrev = 'Pre-Fab Arc', expansion = "base" },
+	-- codex?
+	['????_REDACTED_????'] = { abbrev = 'Scenario Destroyer'},
```

Comment:
REDACTED is `{ type = "unitUpgrade", requirements = "RR",  abbrev = 'Scenario Destroyer', source = "Codex 1" }`
EXCEPTION is `{ type = "Yellow", requirements = "Y", abbrev = 'Scenario Tech', source = "Codex 1" }`